src/fw/startup/system_stm32f4xx.c

```diff
@@ -0,0 +1,405 @@
+/**

```

Comment:
this code is technically part of the HAL, so it should reside together with it.

Reply:
In `src/fw/vendor/CMSIS-ST/Device/ST/STM32F4xx`? In the original distribution it's part of some `source/templates` directory.

Reply:
I guess we could start with using the template as-is. In general, I'd expect startup code to be handled by the OS, not by a random vendor file (even though it _just works_ now, including nRF, so it can be left as is).

Reply:
I should mention, the content is not as-is in the template. All the clock configuration had to be changed in order for the board to boot and I removed a bunch of ifdef'd out stuff we don't use.

Let me know what you want me to change here and I'll make the change. :)

Reply:
Ok, then keep as is!